geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLSpatialRelationshipsTest.scala

```diff
@@ -26,171 +28,360 @@ class SparkSQLSpatialRelationshipsTest extends Specification with LazyLogging {
+      "intEdge" -> "POLYGON(( 0  1,  0  2,  1  2,  1  1,  0  1))",
+      "overlap" -> "POLYGON((-1  1, -1  2,  1  2,  1  1, -1  1))",
+      "extEdge" -> "POLYGON((-1  1, -1  2,  0  2,  0  1, -1  1))",
+      "ext"     -> "POLYGON((-2  1, -2  2, -1  2, -1  1, -2  1))",
+      "corner"  -> "POLYGON((-1 -1, -1  0,  0  0,  0 -1, -1 -1))")
 
     // before
     step {
       ds = DataStoreFinder.getDataStore(dsParams)
       spark = SparkSQLTestUtils.createSparkSession()
       sc = spark.sqlContext
+      SQLTypes.init(sc)
 
-      SparkSQLTestUtils.ingestChicago(ds)
+      SparkSQLTestUtils.ingestPoints(ds, "points", points)
+      dfPoints = spark.read
+        .format("geomesa")
+        .options(dsParams)
+        .option("geomesa.feature", "points")
+        .load()
+      logger.info(dfPoints.schema.treeString)
+      dfPoints.createOrReplaceTempView("points")
+
+      SparkSQLTestUtils.ingestGeometries(ds, "lines", lines)
+      dfLines = spark.read
+        .format("geomesa")
+        .options(dsParams)
+        .option("geomesa.feature", "lines")
+        .load()
+      logger.info(dfLines.schema.treeString)
+      dfLines.createOrReplaceTempView("lines")
 
-      df = spark.read
+      SparkSQLTestUtils.ingestGeometries(ds, "boxes", boxes)
+      dfBoxes = spark.read
         .format("geomesa")
         .options(dsParams)
-        .option("geomesa.feature", "chicago")
+        .option("geomesa.feature", "boxes")
         .load()
-      logger.info(df.schema.treeString)
-      df.createOrReplaceTempView("chicago")
+      logger.info(dfBoxes.schema.treeString)
+      dfBoxes.createOrReplaceTempView("boxes")
     }
 
     // DE-9IM comparisons
-    val box        = "POLYGON(( 0  0,  0 10, 10 10, 10  0,  0  0))"
-
-    val pointInt    = "POINT(5 5)"
-    val pointEdge   = "POINT(0 5)"
-    val pointCorner = "POINT(0 0)"
-    val pointExt    = "POINT(-5 0)"
-    val boxInt     = "POLYGON(( 1  1,  1  2,  2  2,  2  1,  1  1))"
-    val boxIntEdge = "POLYGON(( 0  1,  0  2,  1  2,  1  1,  0  1))"
-    val boxOverlap = "POLYGON((-1  1, -1  2,  1  2,  1  1, -1  1))"
-    val boxExtEdge = "POLYGON((-1  1, -1  2,  0  2,  0  1, -1  1))"
-    val boxExt     = "POLYGON((-2  1, -2  2, -1  2, -1  1, -2  1))"
-    val boxCorner  = "POLYGON((-1 -1, -1  0,  0  0,  0 -1, -1 -1))"
-
-    def testBool(f: String, name: String, g1: String, g2: String, expected: Boolean) = {
-      f+" "+name >> {
-        val sql = s"select $f(st_geomFromWKT('$g1'), st_geomFromWKT('$g2'))"
-        val r = sc.sql(sql)
-        r.collect()
-        r.head.getBoolean(0) mustEqual expected
-      }
+    def testData(sql: String, expectedNames: Seq[String]) = {
+      val r = sc.sql(sql)
+      val d = r.collect()
+      val column = d.map(row => row.getAs[String]("name")).toSeq
+      column must containTheSameElementsAs(expectedNames)
+    }
+
+    def testDirect(f: String, name: String, g1: String, g2: String, expected: Boolean) = {
```

Comment:
Probably should have an inner test block here that prints the name and function.